app/main.py

```diff
@@ -83,33 +80,28 @@ class QuerySequence(BaseModel):
 
 
 class BlastJob(BaseModel):
-    genome_ids: list[str]
+    genome_ids: list[UUID]
     query_sequences: list[QuerySequence]
     parameters: BlastParams
 
-    @validator("genome_ids", each_item=True)  # Check each genome id
-    def check_genome_id(cls, uuid):
-        assert uuid in app.genome_ids, f"{uuid} is not a valid genome ID"
-        return uuid
-
 
 class JobIDs(BaseModel):
     job_ids: list[str]
 
 
-# Infer the target species index file for BLAST payload
-def get_blast_filename(genome_id: str, db_type: str) -> str:
-    suffix = f"{db_type}.toplevel" if db_type.startswith("dna") else f"{db_type}.all"
-    return f"ensembl/{app.genome_ids[genome_id]}/{db_type}/{app.genome_ids[genome_id]}.{suffix}"
+# Infer the target db path for BLAST job
+def get_db_path(genome_id: str, db_type: str) -> str:
+    suffix_map = {"dna": "unmasked", "dna_sm": "softmasked"} #dna/dna_sm/cdna/pep
```

Comment:
Why not global ?

Reply:
Global `suffix_map`? Why?

Reply:
Every time get_db_path is being called it will assign suffix_map with the dictionary so if you make 10 calls it will be assigned 10 times and used in subsequent step. 
If you keep it global it will assigned only once and used however many times

Reply:
Done